easybuild/easyblocks/generic/cargo.py

```diff
@@ -255,6 +255,9 @@ def set_cargo_vars(self):
         env.setvar('RUST_LOG', 'DEBUG')
         env.setvar('RUST_BACKTRACE', '1')
 
+        # Use environment variable since it would also be passed along to builds triggered via python packages
+        env.setvar('CARGO_NET_OFFLINE', 'true')
```

Comment:
This should be set conditional on the cargo easyblock variable "offline" 
```suggestion
        if self.cfg['offline']:
            env.setvar('CARGO_NET_OFFLINE', 'true')
```